datasync/chaindatafetcher/kafka/consumer.go

```diff
@@ -306,37 +307,99 @@ func (c *Consumer) updateOffset(buffer [][]*Segment, lastMsg *sarama.ConsumerMes
 	return nil
 }
 
+// renewExpireMsg updates expiration timer and message if the expiring message is handled.
+func (c *Consumer) renewExpireMsg(buffer [][]*Segment, timer *time.Timer, oldestMsg *sarama.ConsumerMessage) *sarama.ConsumerMessage {
```

Comment:
Does `expire` mean `oldest` here? It looks confusing since you used variable `oldestMsg` here, but what is passed is `expireMsg`. 

Also, this function. does not `renew` a message, instead it resets the timer if the given oldestMsg is handled before this function is called (did I understand correctly?)

Reply:
1. Right. So, actually the oldest message is the candidate to be expired.
2. Yes, it resets or stops the timer if necessary.

I updated the method names and variable. Please take a look.